### ISSUE TITLE
`A::average([])` fails with division by zero

### ISSUE THREAD

**Issue by adamkiss** [k-next/kirby#4269]
## Description

Trying to average an empty array, Kirby throws a "division by zero" error.

**Expected behavior**  
I'd expect either `0` or `null` as return value.

## To reproduce
```php
A::average([]);
```
## Note

I already started a PR that checks for emptiness and returns a 0 - here: https://github.com/getkirby/kirby/blob/3de403cabcd6755bdd95c159bd282ec63ec14658/src/Toolkit/A.php#L642

But then I realised perhaps "zero" is not the way to go, but null is?

**Comment by distantnative** [comment]
`0` would indeed be not correct... `null` would be a breaking change as the method is supposed to return a float so far. What's the use case where the error is not the best result?

**Comment by adamkiss** [comment]
@distantnative 
``` php
$avg = A::average($sometimesEmpty->pluck('averageThis'), 1);
```

Perhaps, third argument `$fallback`?
```php
public static function average(array $array, int $decimals = 0, $fallback = null): float
{
  if (empty($array) && !is_null($fallback)) {
   return $fallback;
  }
  // …
}
```

?

**Comment by lukasbestle** [comment]
From all the options, I think `?float` (with `null` on error) makes the most sense. But yes, it would be a breaking change (only something for 3.7).

**Comment by adamkiss** [comment]
I'd personally prefer the fallback version, but I'll settle for catching errors in a Kirby semi-fluent way

example:
```php
$avg = A::average($sometimesEmpty->pluck('averageThis'), 1) ?? 0;
```

Would it be possible to go with optional `$catch` parameter in the meantime? Too much overhead?

``` php
function average(..., $catch = false) {
  if (!empty($array) && $catch) { return null; }
}
```

That would also be non-breaking

**Comment by distantnative** [comment]
I think then I'd rather leave the method as is and wraps the call in a try-catch block

**Comment by adamkiss** [comment]
@distantnative right, and rethrow in the default case?

**Comment by distantnative** [comment]
No, I meant don't change anything and in the code you use you do

```php
try {
    $avg = A::average($sometimesEmpty->pluck('averageThis'), 1);
} catch ($e) {
    $avg = null;
}
```

**Comment by adamkiss** [comment]
@distantnative that is incredibly ugly, doesn't work for inline stuff (like averaging inside an array assignment), and if you need multiple averages… ugh

**Comment by lukasbestle** [comment]
What Nico suggested is certainly a workaround. I think we should still get this issue fixed in 3.7 with the nullable return value. It's a breaking change, yes. But one that makes sense.

**Comment by bastianallgeier** [comment]
✅
